libr/io/io_desc.c

```diff
@@ -92,11 +92,19 @@ R_API RIODesc *r_io_desc_get_prev(RIO *io, RIODesc *desc) {
 }
 
 R_API RIODesc *r_io_desc_get_highest(RIO *io) {
-	return r_io_desc_get (io, r_io_fd_get_highest (io));
+	int fd = r_io_fd_get_highest (io);
+	if (fd == -1) {
+		return NULL;
+	}
+	return r_io_desc_get (io, fd);
 }
 
 R_API RIODesc *r_io_desc_get_lowest(RIO *io) {
-	return r_io_desc_get (io, r_io_fd_get_lowest (io));
+	int fd = r_io_fd_get_lowest (io);
```

Comment:
I'm sure this is not needed as r_io_desc_get should return NULL if fd is not valid

Reply:
well, it is always good to check for errors after calling a function. I'm very fine with having a `fd == -1` check after getting `fd`.

Reply:
can we make fds signed to handle errors with -1? right now there's an inconsistency between idstorage and iofds, first are ut32 , second are int.

Reply:
they are signed in the context of io. negative values are used to indicate errors by io, not id_storage. id_storage used as io->files is initialized with parameters that ensure that no negative fd can be inserted

Reply:
i think we can discuss the whole rio api in a separate place, not in this PR, i just wanted a single command without having to create new APIs or mess with low level discussions of APIs, which i'm totally happy with them, but i dont think it should block the PR because the discussion goes beyond the actual feature, the new apis and the already defined APIs.